packages/providers/src.ts/base-provider.ts

```diff
@@ -662,11 +672,26 @@ export class BaseProvider extends Provider {
         return logger.throwError("NOT_IMPLEMENTED", Logger.errors.NOT_IMPLEMENTED);
     }
 
+    /* Events, Event Listeners & Polling */
+
+    _startEvent(event: Event): void {
+        this.polling = (this._events.filter((e) => e.pollable()).length > 0);
+        this._previousPollingTimestamps[event.tag] = Timestamp.generate();
```

Comment:
I think it's better to have the polling timestamp in a `Timestamp` object rather than perform a computation in order to derive it (in the `forof events`). It's nothing more than 2 `long`s integers